solara/lab/components/chat.py

```diff
@@ -0,0 +1,198 @@
+            label="Type a message...",
+            v_model=message,
+            on_v_model=set_message,
+            rounded=True,
+            filled=True,
+            hide_details=True,
+            style_="flex-grow: 1;",
+            disabled=disabled,
+        )
+
+        use_change(message_input, send, update_events=["keyup.enter"])
+
+        button = solara.v.Btn(color="primary", icon=True, children=[solara.v.Icon(children=["mdi-send"])], disabled=message == "")
+
+        use_change(button, send, update_events=["click"])
+
+
+@solara.component
+def ChatMessage(
+    children: Union[List[solara.Element], str],
+    user: bool = False,
+    avatar: Union[solara.Element, str, Literal[False], None] = None,
+    name: Optional[str] = None,
+    color: Optional[str] = "rgba(0,0,0,.06)",
+    avatar_background_color: Optional[str] = None,
+    border_radius: Optional[str] = None,
+    notch: bool = False,
+    style: Optional[Union[str, Dict[str, str]]] = None,
+    classes: List[str] = [],
+):
+    """
+    The ChatMessage component renders a message. Messages with `user=True` are rendered on the right side of the screen,
+    all others on the left.
+
+    # Arguments
+
+    * `children`: A list of child components.
+    * `user`: Whether the message is from the current user or not.
+    * `avatar`: An avatar to display next to the message. Can be a string representation of a URL or Material design icon name,
+        a solara Element, False to disable avatars altogether, or None to display initials based on `name`.
+    * `name`: The name of the user who sent the message.
+    * `color`: The background color of the message. Defaults to `rgba(0,0,0,.06)`. Can be any valid CSS color.
+    * `avatar_background_color`: The background color of the avatar. Defaults to `color` if left as `None`.
+    * `border_radius`: Sets the roundness of the corners of the message. Defaults to `None`,
+        which applies the default border radius of a `solara.Column`, i.e. `4px`.
+    * `notch`: Whether to display a speech bubble style notch on the side of the message.
+    * `style`: CSS styles to apply to the component. Either a string or a dictionary. Applied to the container of the message.
+    * `classes`: A list of CSS classes to apply to the component. Applied to the same container.
+    """
+    style_flat = solara.util._flatten_style(style)
+
+    if "border-radius" not in style_flat:
+        style_flat += f"border-radius: {border_radius if border_radius is not None else ''};"
+    if f"border-top-{'right' if user else 'left'}-radius" not in style_flat:
+        style_flat += f"border-top-{'right' if user else 'left'}-radius: 0;"
+    if "padding" not in style_flat:
+        style_flat += "padding: .5em 1.5em;"
+
+    msg_uuid = solara.use_memo(lambda: str(uuid.uuid4()), dependencies=[])
+    with solara.Row(
+        justify="end" if user else "start",
+        style={"flex-direction": "row-reverse" if user else "row", "padding": "5px"},
```

Comment:
Does it make sense to not apply the style at the top level container like we always do?
It's a bit tricky how we deal with the default style that is condition..

Reply:
Ok, we decided to do it like this for now, it's in lab, so it can always change if we feel limited by it.